pontoon/projects/views.py

```diff
@@ -186,22 +188,18 @@ def ajax_notifications(request, slug):
 
     # Detect previously sent notifications using a unique identifier
     # TODO: We should simplify this with a custom Notifications model
-    notifications = []
-
-    identifiers = {
-        data["identifier"]
-        for data in list(
-            Notification.objects.filter(
-                description__isnull=False,
-                target_content_type=ContentType.objects.get_for_model(project),
-                target_object_id=project.id,
-            ).values_list("data", flat=True)
-        )
-    }
-
-    for identifier in identifiers:
-        notifications.append(Notification.objects.filter(data__contains=identifier)[0])
-
+    notifications_map = {}
+
+    for notification in Notification.objects.filter(
+        description__isnull=False,
+        target_content_type=ContentType.objects.get_for_model(project),
+        target_object_id=project.id,
+    ):
+        identifier = notification.data["identifier"]
+        if identifier not in notifications_map:
+            notifications_map[identifier] = notification
+
+    notifications = list(notifications_map.values())
```

Comment:
Brings big performance gains by dropping slow DB calls - one for each past notification.

Reply:
It would be nice if there were a way for Django to generate a `DISTINCT ON (data->'identifier')` select query, but it looks like the postgres type for that field is `text` rather than `jsonb`.

Reply:
Yeah, as mentioned in line 190, `django-notifications` is quite out of date and we should write our own notifications system. This field is not even using the Django's [JSONField](https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.JSONField).